### DISCUSSION TITLE
Enabling Kani Logging

### DISCUSSION THREAD

**Discussion by dcssi** [zhudotexe/kani#17] [Q&A] [answered]
The docs mention that to enable debug logging, we need to declare the KANI_DEBUG as an environment variable.  I did this, but it did not work.  My code is not using the Kani terminal, so the code that does logging.basicConfig is never called.

I wonder if this code 
"
 if os.getenv("KANI_DEBUG") is not None:
        logging.basicConfig(level=logging.DEBUG)
"
should be part of the constructor of kani, rather than in the terminal it self ?
Or its the responsibility of whoever uses the framework without the terminal to init the logging.

**Comment by zhudotexe** [comment, marked as answer]
Good point - the docs only mention this in the context of `chat_in_terminal` since we didn't want kani's loggers to clobber developers' logging setups in a prod environment (since `logging.basicConfig` is fairly nuclear). In a larger application (i.e. not using the `chat_in_terminal` dev utils), you'd want to initialize the logging yourself.

[python.org has a pretty good overview of the logging module](https://docs.python.org/3/howto/logging.html), but depending on your use case, you might be interested in overriding [`Kani.add_to_history`](https://kani.readthedocs.io/en/latest/customization.html#custom-chat-history-updates) to log messages instead.

**Reply by dcssi** [reply]
Totally understandable, I did in my app (that is set up the logging)
